### DISCUSSION TITLE
Error with Alpine and Livewire related with x-show

### DISCUSSION THREAD

**Discussion by skalero01** [alpinejs/alpine#3285] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2640

<div type='discussions-op-text'>

<sup>Originally posted by **skalero01** February  4, 2022</sup>
Hi, i have a big error with alpine and livewire that i havent be able to fix it. 

**The error:** When the page is reloaded with livewire the values on x-show doesnt change, but on @click it works perfectly. Even if i add an id to force the dom to change when we execute alpine the passed variables are the same that the last request.

Basically i have the next code on the html:

```html
<td colspan="2">
    &emsp;
    {{ $platform->platform }}
    <span x-show="shouldShowPlatform('{{$platform->key}}')">
        <i class="fas fa-chevron-up ml-2" role="button" @click="hidePlatform('{{$platform->key}}')"></i>
    </span>
    <span x-show="!shouldShowPlatform('{{$platform->key}}')">
        <i class="fas fa-chevron-down ml-2" role="button" @click="showPlatform('{{$platform->key}}')"></i>
    </span>
</td>
```

And the next code for alpine:

```javascript
function hideTabs() {
    return {
        hidden_deals: [],
        show_platforms: [],

        emptyDeals() {
            this.hidden_deals = [];
        },
        hideDeal(deal) {
            if(Array.isArray(deal)) {
                this.hidden_deals = deal;
                return;
            }
            this.hidden_deals.push(deal);
        },
        showDeal(deal) {
            let index = this.hidden_deals.indexOf(deal);
            if (index > -1) {
                this.hidden_deals.splice(index, 1);
            }
        },
        shouldShowDeal(deal) {
            return !this.hidden_deals.includes(deal)          
        },
        hidePlatform(platform) {
            let index = this.show_platforms.indexOf(platform);
            if (index > -1) {
                this.show_platforms.splice(index, 1);
            }
        },
        showPlatform(platform) {
            this.show_platforms.push(platform);
        },
        expandAll(platforms) {
            this.show_platforms = platforms;                
        },
        emptyPlatforms() {
            this.show_platforms = [];                
        },
        shouldShowPlatform(platform) {
            console.log(this.show_platforms);
            console.log(platform);
            return this.show_platforms.includes(platform)          
        },
        restart() {
            this.emptyPlatforms();
            this.emptyDeals();
        }
    }
}
```

So the first request on the html i get something like this

```html
<td colspan="2">
    &emsp;
    AF
    <span x-show="shouldShowPlatform('4:AF')" style="display: none;">
        <i class="fas fa-chevron-up ml-2" role="button" @click="hidePlatform('4:AF')"></i>
    </span>
    <span x-show="!shouldShowPlatform('4:AF')">
        <i class="fas fa-chevron-down ml-2" role="button" @click="showPlatform('4:AF')"></i>
    </span>
</td>
```

then if i change the filters to get different values i get the next html

```html
<td colspan="2">
    &emsp;
    AS
    <span x-show="shouldShowPlatform('4:AF')" style="display: none;">
        <i class="fas fa-chevron-up ml-2" role="button" @click="hidePlatform('4:AS')"></i>
    </span>
    <span x-show="!shouldShowPlatform('4:AF')">
        <i class="fas fa-chevron-down ml-2" role="button" @click="showPlatform('4:AS')"></i>
    </span>
</td>
```

So as you notice the x-show dom is not updated, so what i did was adding an id on the <td> tag, letting the code like this

```html
<td colspan="2" id="{{$platform->key}}">
    &emsp;
    {{ $platform->platform }}
    <span x-show="shouldShowPlatform('{{$platform->key}}')">
        <i class="fas fa-chevron-up ml-2" role="button" @click="hidePlatform('{{$platform->key}}')"></i>
    </span>
    <span x-show="!shouldShowPlatform('{{$platform->key}}')">
        <i class="fas fa-chevron-down ml-2" role="button" @click="showPlatform('{{$platform->key}}')"></i>
    </span>
</td>
```

After this it worked when inspecting the html, the values were ok, but inside the alpine component it still was receiving the same data as before, as if the change was not done.

So i still have the bug.

Share you a video when you can see the behaviour https://www.loom.com/share/bb960968bd9b4feba89b024ae89ba85b 

I am using the last versions of livewire and alpine. 

Hope you can help me. Regards</div>

**Comment by gone** [comment]
I'm seeing something similar w/ HTMX and an x-show directive. When swapping content out it looks like the initial `display: none` isn't added to the element. I have also set x-cloak and that is getting removed.

I also have `  :class="open ? '' : 'hidden'"` added, and I'm not seeing that binding init either.

**Comment by rubenjonk** [comment]
> I'm seeing something similar w/ HTMX and an x-show directive. When swapping content out it looks like the initial `display: none` isn't added to the element. I have also set x-cloak and that is getting removed.

I am having the same issue. I have a page with a list of objects, which can be expanded for more information. When changing the list by adding or removing an object, I refresh part of the page using HTMX. On refresh, all rows are expanded, while my js variable 'open' is set to false. toggeling the variable true and back closes the expanded details. 

Changing `x-cloak` to `:x-cloak="!open"` (where `open` is my variable to show or hide the element) seems to solve the issue.

**Comment by blakedietz** [comment]
I'm having similar issues with this for Phoenix LiveView. The given example with the up and down chevrons is exactly what's happening when we reproduce this in liveview.

**Comment by skalero01** [comment]
I fixed it adding id="uniqueid" to the divs as i remember, maybe that helps you @blakedietz

**Reply by JTibblesSoftcon** [reply]
This worked for me, thanks

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Comment by JTibblesSoftcon** [comment]
Seems to be an issue if you have more than one "x-show"elements inside once component. You have to give each one a unique Id and it should work
